bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/ui/navigator/CSymbolsContentProvider.java

```diff
@@ -135,6 +182,14 @@ protected void refreshTreeContentFromLS(IFile file) {
 				}
 			}
 		}
+
+	}
+
+	private static ITextFileBuffer getBuffer(IDocument document) {
+		ITextFileBufferManager bufferManager = FileBuffers.getTextFileBufferManager();
+		if (bufferManager == null)
+			return null;
+		return bufferManager.getTextFileBuffer(document);
```

Comment:
There is a performance warning on using this method:

> Note: This method goes through the list of registered buffers and tests whether its document matches the given one. Therefore this method should not be used in performance critical code.

Is this performance critical code? There are lots of places in LSP4E that call this already, so it may not be making things much worse.

Reply:
This should be fixed now, because we don't need the buffer anymore.